### DISCUSSION TITLE
External Dhcp

### DISCUSSION THREAD

**Discussion by hytheam** [ltsp/ltsp#326] [General]
kindly i need help 

i can get clients to boot while using duel network and the dhcp is done by LTSP server 
the problem is when i switch to use External dhcp server by running ltsp dnsmasq and remove the second network 

the clients can get the ip address from the external dhcp but the problem is it wont boot 

all what i get is the following 

![Capture](https://user-images.githubusercontent.com/74302392/98811367-6b74ae00-2429-11eb-87a0-6852b92f72d9.JPG)
 

any ideas whats wrong ?

**Comment by alkisg** [comment]
Did you test with that exact client when using a dual NIC setup, and it worked? Or were you testing with another client, with a different network adapter?

In some cases, undionly.kpxe fails to initialize the network card, but ipxe.pxe might succeed. Try to run the following command on the LTSP server, to replace undionly.kpxe with ipxe.pxe:

```shell
sudo wget https://boot.ipxe.org/ipxe.pxe -O /srv/tftp/ltsp/undionly.kpxe
```

If that solves the problem, then it's a bug in ipxe, and it should be reported there...

**Reply by hytheam** [reply]
this was tested on the same client both vm & pc 
both worked fine with dual NIC setup 


tried the command to replace undionly.kpxe with ipxe.pxe 

but nothing changed 
![Capture](https://user-images.githubusercontent.com/74302392/98820637-16d82f80-2437-11eb-9a9d-8b7dff069d85.JPG)

still not booting

**Comment by alkisg** [comment]
1. What is your DHCP server configuration? Here's an example:
https://ltsp.org/advanced/isc-dhcp-server/

2. On the LTSP server, run this command. Replace enp2s0 with your actual interface name. What is the output?

```shell
/usr/lib/klibc/bin/ipconfig -n enp2s0
```

3. When ipxe boots, press Ctrl+B to get to the command line. Type `autoboot`. When it fails, type `config`. Upload a screenshot from the `config` output screen.

**Reply by hytheam** [reply]
the output for the command 

(IP-Config: ens32 hardware address 00:50:56:9d:c1:35 mtu 1500 DHCP RARP)

config output screen.

![1](https://user-images.githubusercontent.com/74302392/98828797-2f4d4780-2441-11eb-8fad-95964a3497d9.JPG)
![2](https://user-images.githubusercontent.com/74302392/98828838-370cec00-2441-11eb-9626-0f5094c06617.JPG)
![3](https://user-images.githubusercontent.com/74302392/98828868-3ffdbd80-2441-11eb-87ce-b9a16f6ca97b.JPG)
![4](https://user-images.githubusercontent.com/74302392/98828906-468c3500-2441-11eb-9d87-366830b6c139.JPG)

**Reply by hytheam** [reply]
also in the isc-dhcp-server example configuration file 

```
# This is the LTSP subnet declaration
subnet 192.168.67.0 netmask 255.255.255.0 {
  range 192.168.67.20 192.168.67.250;
  option ipxe.no-pxedhcp 1;
  option routers 192.168.67.1;
  # On single-NIC setups, usually routers != next-server (=TFTP server)
```

according to the install instruction  the external dhcp is just giving ip address 
and no configuration for tftp or next server 

is that right or should i add path of tftp server and location of image in the external dhcp server ?

**Comment by alkisg** [comment]
Regarding the external DHCP config, there are two options. The easiest one is to do proxyDHCP, in which case you don't need to modify anything at all in your DHCP server. But it must not provide a boot filename itself. The second option would be to completely configure the external DHCP server like in the example isc-dhcp-server link.

Regarding the `ipconfig -n enp2s0` command, it looks like the DHCP server didn't reply to the LTSP server. Do you have a firewall between them?

Also, what's the output of this command on the LTSP server? Does it have your current subnet? If not, you need to run `ltsp dnsmasq` to update the dnsmasq configuration:

```shell
grep range /etc/dnsmasq.d/ltsp-dnsmasq.conf
```

After doing any necessary changes, run `journalctl -fu dnsmasq` on the LTSP server, and then try to boot a client. Do you see lines like the following? If not, then it's a firewall/range problem:

```shell
Οκτ 10 12:45:09 alkis dnsmasq-dhcp[1046]: 174336722 PXE(enp2s0) 00:11:09:64:2a:d2 proxy
```

**Reply by hytheam** [reply]
ok 
i think we are getting something 

the output of ( grep range /etc/dnsmasq.d/ltsp-dnsmasq.conf ) 
is 
#IP ranges to hand out, usually on the internal LTSP subnet of 2-NIC setups
dhcp-range=192.168.67.20,192.168.67.250,12h
#If another DHCP server is present on the network, a proxy range may be used
dhcp-range=set:proxy,10.1.0.0,proxy,255.255.254.0

the first dhcp -range is the wrong ips 

the second is the right 

& 

output of journalctl -fu dnsmasq
is 
![6](https://user-images.githubusercontent.com/74302392/98834293-b3a2c900-2447-11eb-8970-0d3579eecab4.JPG)

the error 0 TFTP Aborted received from 10.1.0.89

that ip is for the client i am trying to boot

**Comment by alkisg** [comment]
The DHCP ranges are OK. The TFTP error isn't a problem.

Did you manage to get a reply from the DHCP server when running `/usr/lib/klibc/bin/ipconfig -n ens32` on the LTSP server?

**Reply by hytheam** [reply]
the replay is 

IP-Config: ens32 hardware address 00:50:56:9d:c1:35 mtu 1500 DHCP RARP

**Comment by alkisg** [comment]
You need to fix your DHCP server or firewall in order to get a reply. For example:

```shell
root@alkis:~# /usr/lib/klibc/bin/ipconfig -n enp1s0
IP-Config: enp1s0 hardware address e4:54:e8:d7:d9:12 mtu 1500 DHCP RARP
IP-Config: enp1s0 guessed broadcast address 192.168.1.255
IP-Config: enp1s0 complete (dhcp from 192.168.1.1):
 address: 192.168.1.3      broadcast: 192.168.1.255    netmask: 255.255.255.0   
 gateway: 192.168.1.1      dns0     : 192.168.1.1      dns1   : 0.0.0.0         
 rootserver: 192.168.1.1 rootpath: 
 filename  : 
```
